rmc/src/lib.rs

```diff
@@ -1,619 +1,34 @@
-//! Reliable MultiCast - a primitive for Reliable Broadcast protocol.
 pub use aleph_bft_crypto::{
     Indexed, MultiKeychain, Multisigned, NodeCount, PartialMultisignature, PartiallyMultisigned,
     Signable, Signature, Signed, UncheckedSigned,
 };
-use async_trait::async_trait;
 use codec::{Decode, Encode};
 use core::fmt::Debug;
-use futures::{
-    channel::mpsc::{unbounded, UnboundedReceiver, UnboundedSender},
-    future::pending,
-    FutureExt, StreamExt,
-};
-use futures_timer::Delay;
-use log::{debug, warn};
-use std::{
-    cmp::Reverse,
-    collections::{BinaryHeap, HashMap},
-    fmt::Formatter,
-    hash::Hash,
-    ops::{Add, Div, Mul},
-    time::{Duration, Instant},
-};
+use std::hash::Hash;
 
-/// Abstraction of a task-scheduling logic
-///
-/// Because the network can be faulty, the task of sending a message must be performed multiple
-/// times to ensure that the recipient receives each message.
-/// The trait [`TaskScheduler<T>`] describes in what intervals some abstract task of type `T`
-/// should be performed.
-#[async_trait::async_trait]
-pub trait TaskScheduler<T>: Send + Sync {
-    fn add_task(&mut self, task: T);
-    async fn next_task(&mut self) -> Option<T>;
-}
+mod handler;
+mod scheduler;
+mod service;
+
+pub use handler::Handler;
+pub use scheduler::DoublingDelayScheduler;
+pub use service::Service;
 
-/// An RMC message consisting of either a signed (indexed) hash, or a multisigned hash.
+/// An RMC hash consisting of either a signed (indexed) hash, or a multisigned hash.
 #[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, Hash)]
-pub enum Message<H: Signable, S: Signature, M: PartialMultisignature> {
+pub enum RmcMessage<H: Signable, S: Signature, M: PartialMultisignature> {
```

Comment:
Well, this rename looks kinda silly...

Reply:
actually it was `Message --> RmcHash --> RmcMessage` :stuck_out_tongue:
idk, I like `RmcMessage` more because we don't have to make aliases for it in `alerter` or use the full path `aleph_bft_rmc::Message`. Just import `RmcMessage` and use as it is.